### PR TITLE
0.1 + 0.2 returns correct result with use of round()

### PR DESCRIPTION
**Pull request by grimxreaper**
0.1 + 0.2 returns the correct result in the format we want with round(), for the uses of this app we want to remove the extra decimal places when displaying the result, so we are rounding the result.

Issue #4 

Links to SO examples I used to understand the problem:
[link 1](https://stackoverflow.com/questions/588004/is-floating-point-math-broken)
[link 2](https://stackoverflow.com/questions/1458633/how-to-deal-with-floating-point-number-precision-in-javascript)
[link 3](https://docs.python.org/release/2.5.1/tut/node16.html)

### CODE REVIEW

**Comment by bryzaguy** [comment]
These PRs should have titles and descriptions that help me understand what they intend to do. A few recommendations: 1) I like that you have "bug/***" for your branch name. I recommend naming the branch related to the fix, rather than Issue number. For example: "bug/incorrect-fraction-addition". 2) Some of the commit messages need more detail. A good rule of thumb for commit messages is they complete the sentence "Applying this commit will _____".

File: src/App.js [hunk 1]
```diff
@@ -117,10 +102,17 @@ class App extends React.Component {
       }
     }
 
-    console.log("here");
     if ("0123456789".includes(result[result.length - 1])) {
       console.log("checking after digit");
-      console.log(result[result.length - 1]);
+      console.log(result[result.length - 1]); //1
```
Comment (bryzaguy) [inline src/App.js:107]: Console logs should be removed before we can merge this.

File: src/App.test.js [hunk 2]
```diff
@@ -55,6 +55,17 @@ test("displays correct result of adding -7 to 7", () => {
   expect(evaluate('7+-7')).toBe(0);
 });
 
+test("displays correct result of adding 0.1 and 0.2", () => {
```
Comment (bryzaguy) [inline src/App.test.js:58]: Why doesn't this test use fireEvent and render App like the others?

File: src/App.test.js [hunk 3]
```diff
@@ -209,26 +200,26 @@ describe('error message displaying at correct times', () => {
 //   })
 // }
 
-test('equal button is called one time', () => {
-  const x = 3;
-  const y = 4;
-  const onSubmit = jest.fn();
-  render(<App onSubmit={"="} />);
-  const { getByTestId } = render(<App />);
+// test('equal button is called one time', () => {
```
Comment (bryzaguy) [inline src/App.test.js:226]: Why not delete this code rather than commenting it out?

File: src/App.test.js [hunk 4]
```diff
@@ -192,6 +192,18 @@ describe('error message displaying at correct times', () => {
 
     expect(getByTestId("result")).toHaveTextContent("21")
   });
+
+  test("returns 0 instead of an empty string", () => {
```
Comment (bryzaguy) [inline src/App.test.js:207]: This looks unrelated to the bug.

File: src/App.js [hunk 5]
```diff
@@ -120,14 +121,18 @@ class App extends React.Component {
 // - operator
 // - ), only if there are more ( than )
 // .
+    console.log("here");
```
Comment (bryzaguy) [inline src/App.js]: these console logs should be removed.

File: src/App.test.js (hunk 4 above)
Comment (grimxreaper) [reply src/App.test.js:207]: I had forgotten to change the wording for what this test is testing.

Comment (grimxreaper) [reply src/App.test.js:207]: Fixed the wording.

File: src/App.test.js (hunk 3 above)
Comment (grimxreaper) [reply src/App.test.js]: Yes! Fixed.

**Review by grimxreaper — COMMENTED** [review]
Fixed & replied.

File: src/App.test.js (hunk 2 above)
Comment (grimxreaper) [reply src/App.test.js]: Ah yes, that was test code that was being used to check if evaluate was working, and to check if using a config for math.js would help with the decimal bug. I removed it as it is no longer necessary.

**Comment by grimxreaper** [comment]
Links to SO examples I used to understand the problem:
[](https://stackoverflow.com/questions/588004/is-floating-point-math-broken)
[](https://stackoverflow.com/questions/1458633/how-to-deal-with-floating-point-number-precision-in-javascript)
[](https://docs.python.org/release/2.5.1/tut/node16.html)

File: src/App.js [hunk 6]
```diff
@@ -92,32 +89,21 @@ class App extends React.Component {
...
       if (result.length > 1) {
         //these few lines below, did not work to fix another bug
-       console.log('here')
         if ("0123456789".includes(result[result.length - 1])) {
           //remove the previous number and return the last number
           return result + button.replace("(", "*(");
         }
       }
     }
+
```
Comment (bryzaguy) [inline src/App.js:106]: let's minimize the diff by getting rid of the added blank lines

File: src/App.test.js (hunk 4 above)
Comment (bryzaguy) [reply src/App.test.js:185]: Are the changes pushed?

**Comment by bryzaguy** [comment]
What you have as a description is really close to a good title! Let's update the title and description, like the other PRs.

Thanks for adding the SO links. Would be a good idea to update the description of the PR with those links.  I think I like this answer the best, since rounding is a hack-ish workaround that could get you into trouble if you needed those numbers to be exact: https://stackoverflow.com/a/3439981

Typically it's good to do this kind of research (the SO answers) and propose it before writing any code.

File: src/App.test.js (hunk 4 above)
Comment (grimxreaper) [reply src/App.test.js:185]: Yes
